sw/device/tests/entropy_src_fw_observe_many_contiguous.c

```diff
@@ -103,9 +108,63 @@ static status_t let_observe_fifo_overflow(uint32_t timeout_usec) {
   return OK_STATUS();
 }
 
+void ottf_external_isr(uint32_t *exc_info) {
+  dif_rv_plic_irq_id_t plic_irq_id;
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&rv_plic, kTopEarlgreyPlicTargetIbex0,
+                                     &plic_irq_id));
+
+  top_earlgrey_plic_peripheral_t peripheral =
+      top_earlgrey_plic_interrupt_for_peripheral[plic_irq_id];
+  CHECK(peripheral == kTopEarlgreyPlicPeripheralEntropySrc);
+
+  CHECK(plic_irq_id == kTopEarlgreyPlicIrqIdEntropySrcEsObserveFifoReady);
+
+  CHECK(!entropy_src_fifo_has_overflowed(),
+        "observe FIFO overflowed during collection");
+
+  uint32_t buf[32];
+  size_t len = ARRAYSIZE(buf);
+  CHECK_DIF_OK(
+      dif_entropy_src_observe_fifo_nonblocking_read(&entropy_src, buf, &len));
+
+  if (len > 0) {
+    CHECK_DIF_OK(
```

Comment:
Maybe add a comment explaining why it is ok to do this operation in the IRQ handler?

Reply:
Do you mean justify why writing to the FIFO will not block the IRQ for too long?